app/presenters/creator_home_presenter.rb

```diff
@@ -65,7 +65,10 @@ def creator_home_props
       end
     end
 
-    previous_year = Time.current.prev_year.year
+    tax_forms = (Time.current.year.downto(seller.created_at.year)).each_with_object({}) do |year, hash|
+      url = seller.eligible_for_1099?(year) ? seller.tax_form_1099_download_url(year: year) : nil
```

Comment:
If the tax form isn't present, it will raise an exception from here: https://github.com/antiwork/gumroad/blob/cbe43b8326a8a767fd0d5203daa2c553a0fb6af1/app/helpers/signed_url_helper.rb#L16-L17

Maybe we can list only the years which have associated tax forms?

Reply:
Hmm, I thought I already was listing only the years with associated tax forms. I only add the year to the map`if url.present?`, so the current code should already account for this.

The rescue statement in the `User.tax_form_1099_download_url(year)` function would swallow the S3 error and simply return `nil` if the tax form isn't present, right?
https://github.com/antiwork/gumroad/blob/717d1d3d0bdec56547780741d5c5a855fa5f64eb/app/models/user.rb#L926-L927

Here's an example where i'm considering years 2020-2025 in the loop, but raising an error when getting the file from s3 unless the file contains "2023". As expected, only 2023 shows up in the dropdown.
https://github.com/user-attachments/assets/0d5bb19b-632c-43d6-a5b5-6e52f77d0e53